test/mind_the_gapp/arrival_time_test.exs

```diff
@@ -0,0 +1,75 @@
+defmodule MindTheGapp.ArrivalTimeTest do
+  use MindTheGapp.DataCase, async: true
+  alias MindTheGapp.Mta.ArrivalTime
+
+  setup_all :create_fake_arrival_time_data
+
+  def create_fake_arrival_time_data(_context) do
+    clear_database()
```

Comment:
This is usually taken care of by the `DataCase`. It should be opening a transaction and then rolling it back for every test. Can we check on that?

Reply:
This goes for the other comments concerning DataCase, but there's an error with testing migrations in Phoenix/Ecto. In 2018, Ecto was updated and the team changed how sandbox interacts with migrations/schemas. 

In summary, I thought so too, but I was getting ownership errors. The thread that lead me to do this, remove `Ecto.Adapters.SQL.Sandbox.mode(MindTheGapp.Repo, :manual)` and add `mix deps.update db_connection` to travis is here: [Issue thread](https://github.com/elixir-ecto/ecto/issues/2782)

Reply:
Oh i see. Thanks for that context.

I poked around at this locally, and I think I understand why this is happening. The `DataCase` will checkout a connection during the `setup` phase, but I think you saw this error because you are seeding data during the `setup_all` phase. If you change the `setup_all` to `setup` and add this like back into `test/test_helper.exs`, it should work.

```
Ecto.Adapters.SQL.Sandbox.mode(MindTheGapp.Repo, :manual)
```

My guess is that `setup_all` is called before any `setup` callbacks are run. You were getting ownership errors because the process your test was running in had not yet checkout a connection from the sandbox. 

In practice, I very rarely use a `setup_all` or `after_all` hook in my tests.

Reply:
I did this locally and got all of the tests to pass without that error.